### PR TITLE
Sanjana/endpoints

### PR DESCRIPTION
**Pull request by skaundinya15**
sample endpoint

I've attached the sample endpoint that I tested it with from the git issue. The way I constructed this as @Jma353 talked about on Sunday was that each route is bound by walking. So from the current location, the user walks, and from the end location the user walks. Further more, I thought it might be useful to redefine how the type place is used. So in my changes I made it such that the end destination is always going to be a place, even if the end destination is just a bus stop. I figured this might be easier to understand as otherwise it would be hard to distinguish if a user wants their end location to be a stop. As in the example endpoint in the git issue, the end location is a bus stop, but in my implementation I've treated it as a place so that it's indicative that the route has ended.

### CODE REVIEW

**Comment by Jma353** [comment]
In the tested example, it labels "Seneca @ Commons" as a "place," but it's actually a "stop."

File: src/routers/GetRouteRouter.js [hunk 1]
```diff
@@ -91,6 +92,37 @@ class GetRouteRouter extends AppDevRouter {
       }
     }
 
+     const stopSummary = [];
+     for (let i = 0; i <result.length; i++) {
+       const r = result[i];
+       const obj = {};
```
Comment (Jma353) [inline src/routers/GetRouteRouter.js]: Better variable name

File: src/routers/GetRouteRouter.js [hunk 2]
```diff
@@ -91,6 +92,37 @@ class GetRouteRouter extends AppDevRouter {
       }
     }
 
+     const stopSummary = [];
```
Comment (Jma353) [inline src/routers/GetRouteRouter.js]: Overall comment: these different string should be constants.  "walk" should be a constant, "place" should be a constant, etc.  If we need to change these every, we'd have to go back through the code and change the use of every one of these plain-text strings.

File: src/routers/GetRouteRouter.js [hunk 3]
```diff
@@ -91,6 +92,37 @@ class GetRouteRouter extends AppDevRouter {
...
+     for (let i = 0; i <result.length; i++) {
+       const r = result[i];
+       const obj = {};
+      if (i == 0 && r.startStop.name == "START_WALKING") {
```
Comment (Jma353) [inline src/routers/GetRouteRouter.js]: ===, not ==

File: src/routers/GetRouteRouter.js [hunk 4]
```diff
@@ -79,6 +79,7 @@ class GetRouteRouter extends AppDevRouter {
     const arrivalTime = Math.floor(leaveBy + (endTime - startTime));
 
     // Then build up stop ordering / walking information
+
```
Comment (Jma353) [inline src/routers/GetRouteRouter.js]: Remove

File: src/routers/GetRouteRouter.js [hunk 5]
```diff
@@ -91,6 +92,37 @@ class GetRouteRouter extends AppDevRouter {
       }
     }
 
+     const stopSummary = [];
+     for (let i = 0; i <result.length; i++) {
+       const r = result[i];
```
Comment (Jma353) [inline src/routers/GetRouteRouter.js]: Better variable name

File: src/routers/GetRouteRouter.js [hunk 6]
```diff
@@ -91,6 +92,37 @@ class GetRouteRouter extends AppDevRouter {
       }
     }
 
+     const stopSummary = [];
+     for (let i = 0; i <result.length; i++) {
```
Comment (Jma353) [inline src/routers/GetRouteRouter.js]: `i < result.length`

File: src/routers/GetRouteRouter.js [hunk 7]
```diff
@@ -91,6 +92,37 @@ class GetRouteRouter extends AppDevRouter {
...
+        obj.name = "Current Location";
+        obj.type = "currentLocation";
+        obj.nextDirection = "walk";
+        obj.busNumber = r.busNum;
+      } else {
+        obj.name = r.startStop.name;
+        obj.busNumber = r.busNum;
+      }
+      if ((i + 1 < result.length) && (obj.nextDirection == null)) {
+        if ((r.busNum != -1) && (r.busNum != null)) {
```
Comment (Jma353) [inline src/routers/GetRouteRouter.js]: `if (r.busNum && r.busNum !== -1) { ... }`

**Comment by Jma353** [comment]
Overall comment - I was expecting you to use my `mainStops` / `mainStopNums` as a tool to format this data, as those two data structures are extremely organized and linear in their structure.  Right now this code has a lot of `if`-statement base conditions that are prone to edge-case scenarios.  I'd recommend refactoring to use the two arrays constructed.

**Comment by skaundinya15** [comment]
Regarding the last stop being labeled as a place, I thought about it and realized that I couldn't find a way to distinguish between a place or a stop for the very end. In other words, if the last point is a stop, I can't tell that it's a stop as I only know the lat and long of that location. So I thought it might be useful for the frontend to label the last stop as a place as thats where the user will be getting off so even if they get off at a stop, it would just be labeled as a place because thats their end destination. I was just thinking this could be a way to do it but if you think there's another way I can distinguish between those two, let me know

**Comment by skaundinya15** [comment]
screen shot 2017-09-07 at 1 28 25 am

Here's the updated screenshot with the final changes I have. @Jma353 I know we talked about using your data structures to do some post processing, so I used them in if conditions to check for certain conditions to return the array of JSON properly. I think these changes will work for the time being; in the future when we change the way we bound routes, we can revisit this to make changes accordingly. However for now I think it's good to get it shipped out now so we can meet our deadline.

**Pull request merged**